### PR TITLE
Lazily evaluate template name for context variable injection

### LINKED ISSUE #461: Liquid variable with include tag and using for/with not working correctly
If you do {% include liquid_var_name %} it loads the template correctly.

But if you do {% include liquid_var_name for arrays %}  it will safe the arrays in the wrong variable name instead of the template it will save it in the iquid_var_nam variable.

### PR DESCRIPTION
**Pull request by pushrax**
Fixes #461, as an alternative to #556. The added tests explain what this allows.

### CODE REVIEW

File: lib/liquid/tags/include.rb [hunk 1]
```diff
@@ -44,7 +43,10 @@ def parse(tokens)
 
     def render(context)
       partial = load_cached_partial(context)
-      variable = context.evaluate(@variable_name)
+
+      template_name = context.evaluate(@template_name_expr)
+      variable = context.evaluate(@variable_name_expr || Expression.parse(template_name))
```
Comment (fw42) [inline lib/liquid/tags/include.rb]: Hm. As I said on the other PR, do we really have to do this here? Calling `parse` in the `render` step feels kinda wrong.

Comment (pushrax) [inline lib/liquid/tags/include.rb]: Unfortunately yes, since `template_name` comes from the context :(

Comment (pushrax) [inline lib/liquid/tags/include.rb]: Wait, I was wrong, it can be improved. Pushed a fix.

File: lib/liquid/tags/include.rb [hunk 2]
```diff
@@ -44,7 +43,10 @@ def parse(tokens)
 
     def render(context)
       partial = load_cached_partial(context)
-      variable = context.evaluate(@variable_name)
+
+      template_name = context.evaluate(@template_name_expr)
+      variable = @variable_name_expr ? context.evaluate(@variable_name_expr) : context[template_name]
```
Comment (fw42) [inline lib/liquid/tags/include.rb]: well that's kinda cheating ;-) `Context[]` just calls `Expression#parse` now :-)

Comment (pushrax) [inline lib/liquid/tags/include.rb]: Right... the only solution then is to create a VariableLookup manually or do what I had before..

**Comment by pushrax** [comment]
@fw42 how about this? Means that you can't do `{{ include "foo.bar" }}` and have it actually look up `foo.bar` but that should be fine?

**Comment by fw42** [comment]
@dylanahsmith mind taking a look as well?

File: lib/liquid/tags/include.rb [hunk 3]
```diff
@@ -44,7 +43,15 @@ def parse(tokens)
 
     def render(context)
       partial = load_cached_partial(context)
-      variable = context.evaluate(@variable_name)
+
+      template_name = context.evaluate(@template_name_expr)
+      context_variable_name = template_name.split('/'.freeze).last
```
Comment (fw42) [inline lib/liquid/tags/include.rb]: where did the `[1..-2]` go?

**Comment by fw42** [comment]
Code looks good to me I guess

File: lib/liquid/tags/include.rb (hunk 3 above)
Comment (pushrax) [inline lib/liquid/tags/include.rb]: `Expression.parse` takes care of turning quoted strings into strings (if it was unquoted it'd return a `VariableLookup` and `context.evaluate` would look up the variable on the previous line).

**Comment by dylanahsmith** [comment]
LGTM

**Pull request merged**